src/components/table-view/cell-editor.jsx

```diff
@@ -258,6 +258,7 @@ class CellEditor extends React.Component {
         }
       }
     }
+    this.forceUpdate();
```

Comment:
This fixes the field type change not showing up in the cell editor (see video of before and after).